### PR TITLE
Sort Tags Alphabetically

### PR DESCRIPTION
**Pull request by njd9419**
This PR adds sorting to the `Tags` component so that it displays them alphabetically by name.

### CODE REVIEW

**Comment by benvogler** [comment]
I think you should do this on the backend as well since there are a lot of places we render tags/materials. I would keep the sort on the frontend since it happens after we combine materials and tags.

File: src/main/java/com/monumental/controllers/TagController.java [hunk 1]
```diff
@@ -30,14 +32,19 @@
...
+            tags = this.tagRepository.getAllByName(name);
         } else if (names != null && names.size() > 0) {
-            return this.tagRepository.getAllByNameIn(names);
+            tags = this.tagRepository.getAllByNameIn(names);
         } else {
-            return this.tagRepository.findAll();
+            tags = this.tagRepository.findAll();
         }
+
+        tags.sort(Comparator.comparing(Tag::getName));
```
Comment (njd9419) [inline src/main/java/com/monumental/controllers/TagController.java]: Java 8 has some neat stuff like this.

**Comment by njd9419** [comment]
I added a sort to the `/api/tags` endpoint. I think that should cover 99% of our bases.

File: src/main/java/com/monumental/controllers/TagController.java (hunk 1 above)
Comment (benvogler) [reply src/main/java/com/monumental/controllers/TagController.java]: I'm trying to decide if this is better or worse than the Javascript equivalent:
```
tags.sort((a, b) => b.getName() - a.getName());
// or if sorting an array of strings
tags.sort();
```

File: client/src/components/Tags/Tags.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ export default class Tags extends React.Component {
         if (!tags) return (<div/>);
         return (
             <div className="tags">
-                {tags.map(tag => {
+                {tags.sort((a, b) => (a.name > b.name) ? 1 : -1).map(tag => {
```
Comment (benvogler) [inline client/src/components/Tags/Tags.js]: speaking of which, you could use that here
```
tags.sort((a, b) => b.name - a.name)
```

Comment (njd9419) [reply client/src/components/Tags/Tags.js]: I'll do that, it's much cleaner.

Comment (njd9419) [reply client/src/components/Tags/Tags.js]: Actually that doesn't seem to work for me.

Comment (benvogler) [reply client/src/components/Tags/Tags.js]: yea I'm dumb, that just returns NaN

**Pull request merged**